src/CabanaPD_Input.hpp

```diff
@@ -104,7 +89,73 @@ class Inputs
         // Not yet a user option.
         inputs["half_neigh"]["value"] = false;
     }
-    ~Inputs() {}
+
+    void setupSize()
+    {
+        if ( inputs.contains( "system_size" ) )
+        {
+            auto size = inputs["system_size"]["value"];
+            std::string size_unit = inputs["system_size"]["unit"];
+            for ( std::size_t d = 0; d < size.size(); d++ )
+            {
+                double s = size[d];
+                double low = -0.5 * s;
+                double high = 0.5 * s;
+                inputs["low_corner"]["value"][d] = low;
+                inputs["high_corner"]["value"][d] = high;
+            }
+            inputs["low_corner"]["unit"] = size_unit;
+            inputs["high_corner"]["unit"] = size_unit;
+        }
+        else if ( inputs.contains( "low_corner" ) &&
+                  ( inputs.contains( "high_corner" ) ) )
+        {
+            auto low_corner = inputs["low_corner"]["value"];
+            auto high_corner = inputs["high_corner"]["value"];
+            std::string size_unit = inputs["low_corner"]["unit"];
```

Comment:
It is possible that the user accidentally includes a different unit for low_corner and high_corner. Should we output an error in that case?

Reply:
Added